include/s3select_functions.h

```diff
@@ -820,6 +823,306 @@ struct _fn_to_timestamp : public base_function
 
 };
 
+struct _fn_to_string : public base_function
+{
+  value val_timestamp;
+  value val_format;
+
+  std::regex yyyy{"yyyy+"};
+  std::regex yyy{"yyy"};
+  std::regex yy{"yy"};
+  std::regex y{"y"};
+  std::regex month_mmmmm{"MMMMM"};
+  std::regex month_mmmm{"MMMM"};
+  std::regex month_mmm{"MMM"};
+  std::regex month_mm{"MM"};
+  std::regex month_m{"M"};
+  std::regex dd{"dd"};
+  std::regex d{"d"};
+  std::regex hour_h2{"HH"};
+  std::regex hour_h1{"H"};
+  std::regex hour_hh{"hh"};
+  std::regex hour_h{"h"};
+  std::regex mint_mm{"mm"};
+  std::regex mint_m{"m"};
+  std::regex sec_ss{"ss"};
+  std::regex sec_s{"s"};
+  std::regex frac_sec{"S+"};
+  std::regex nano_sec{"n"};
+  std::regex meridies{"a"};
+
+  std::vector<std::string> months = { "January", "February", "March","April",
+                        "May", "June", "July", "August", "September",
+                        "October", "November", "December"};
+
+  uint32_t length_same_char_str(std::string str, char ch)
+  {
+    int i = 0;
+    while(str[i] == ch)
+      i++;
+    return i;
+  }
+
+  bool operator()(bs_stmt_vec_t* args, variable* result) override
+  {
+    auto iter = args->begin();
+    int args_size = args->size();
+
+    if (args_size < 2)
+    {
+      throw base_s3select_exception("to_string need 2 parameters");
+    }
+
+    base_statement* dt1_param = *iter;
+    val_timestamp = dt1_param->eval();
+
+    if (val_timestamp.is_timestamp() == false)
+    {
+      throw base_s3select_exception("first parameter should be timestamp");
+    }
+
+    iter++;
+    base_statement* format = *iter;
+    val_format = format->eval();
+
+    if (val_format.is_string() == false)
+    {
+      throw base_s3select_exception("second parameter should be string");
+    }
+
+    boost::posix_time::ptime new_ptime = *val_timestamp.timestamp();
+    std::string fmt = val_format.to_string();
+    std::vector<std::string> res;
+
+    for (uint32_t i = 0; i < fmt.length(); i++)
```

Comment:
OK, it pass the format string one-time.
per each meta-char been found it will search for best-match and push result to res(result sstring)
in the case of constant string(the common case) we wish to avoid that, no need for search, should only print per each new row(value).

/res.push_back/ should decouple from search flow, and replace by vector<print-function> 
thus, for dynamic use-case, the vector<print-function> will re-built per new value (could optimize that also)
but for the constant format , need only to execute the vector<print-function>.